### PR TITLE
stm32cubemx: add working desktop file

### PR DESCRIPTION
**Pull request by prtzl**
Added manual desktop file contents as the builtin version does not work (no file created in `$out/share/applications`).
This is probably not a final solution since a (not working implementation of) builtin function could be fixed.

###### Things done

Create `$out/share/applications/stm32cubemx.desktop` file and fill the contents in *buildCommand*. This works on nixos and on non-nixos (Cinnamon 5.2.7, Gnome 42).

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by wucke13 — APPROVED** [review]

File: pkgs/development/embedded/stm32/stm32cubemx/default.nix [hunk 1]
```diff
@@ -52,6 +54,19 @@ stdenv.mkDerivation rec {
           $out/share/icons/hicolor/"$size"x"$size"/apps/${pname}.png
       fi
     done;
+
+    cat << EOF > $out/share/applications/stm32cubemx.desktop
```
Comment (ners) [inline pkgs/development/embedded/stm32/stm32cubemx/default.nix:58]: I would rather see these contents live in a separate `stm32cubemx.desktop` file, which you can then link via string interpolation.

**Comment by Janik-Haag** [comment]
@prtzl are you still interested in getting this done?

File: pkgs/development/embedded/stm32/stm32cubemx/default.nix (hunk 1 above)
Comment (prtzl) [reply pkgs/development/embedded/stm32/stm32cubemx/default.nix]: Used `makeDesktopItem` instead.

**Comment by prtzl** [comment]
I tried to get it working with `copyDesktopItems` but had no luck. Ended up using `makeDesktopItem` and copying the generated file to the correct directory in $out.

**Comment by wucke13** [comment]
LGTM

**Review by Janik-Haag — APPROVED** [review]

**Pull request merged**